packages/platforms/react-native/ios/BugsnagReactNativePerformance.mm

```diff
@@ -129,6 +138,29 @@ @implementation BugsnagReactNativePerformance
     return config;
 }
 
+RCT_EXPORT_BLOCKING_SYNCHRONOUS_METHOD(startNativeSpan:(NSString *)name
+                options:(NSDictionary *)options) {
+
+    // native spans are always first class and should never become the current context
+    NSMutableDictionary *spanOptions = [options mutableCopy];
+    spanOptions[@"makeCurrentContext"] = [NSNumber numberWithBool:NO];
+    spanOptions[@"firstClass"] = [NSNumber numberWithBool:YES];
+    
+    BugsnagPerformanceSpan *nativeSpan = [BugsnagCocoaPerformanceFromBugsnagReactNativePerformance.sharedInstance startSpan:name options:spanOptions];
+    [nativeSpan.attributes removeAllObjects];
+
+    NSMutableDictionary *span = [NSMutableDictionary new];
+    span[@"name"] = nativeSpan.name;
+    span[@"id"] = [NSString stringWithFormat:@"%llx", nativeSpan.spanId];
+    span[@"traceId"] = [NSString stringWithFormat:@"%llx%llx", nativeSpan.traceId.hi, nativeSpan.traceId.lo];
+    span[@"startTime"] = [NSNumber numberWithDouble: [nativeSpan.startTime timeIntervalSince1970] * NSEC_PER_SEC];
```

Comment:
do we also need to add the first class property, or does that only matter for the native side?

Reply:
The first class span option is always true for native spans (set a bit further up), but the actual attribute will be set in JS (the JS code that calls this hasn't been written yet)